File: consumer/src/main/java/no/nav/sbl/dialogarena/soknadinnsending/consumer/dkif/dto/DigitalKontaktinfoBolk.java
package no.nav.sbl.dialogarena.soknadinnsending.consumer.dkif.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class DigitalKontaktinfoBolk {

    private List<DigitalKontaktinfo> kontaktinfo;
    private List<Feil> feil;

    @JsonCreator
    public DigitalKontaktinfoBolk(
            @JsonProperty("kontaktinfo") List<DigitalKontaktinfo> kontaktinfo,
            @JsonProperty("feil") List<Feil> feil) {
        this.kontaktinfo = kontaktinfo;
        this.feil = feil;
    }

    public List<DigitalKontaktinfo> getKontaktinfo() {
        return kontaktinfo;
    }

    public List<Feil> getFeil() {
        return feil;
    }

    @Override
    public String toString() {
        return "DigitalKontaktinfoBolk{" +
                "kontaktinfo=" + kontaktinfo +
                ", feil=" + feil +
                '}';
    }
}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/client/dkif/dto/DigitalKontaktinfoBolk.kt
package no.nav.sosialhjelp.soknad.client.dkif.dto

data class DigitalKontaktinfoBolk(
    val kontaktinfo: Map<String, DigitalKontaktinfo>?,
    val feil: Map<String, Feil>?
)

data class DigitalKontaktinfo(
    val mobiltelefonnummer: String?
)

data class Feil(
    val melding: String?
)
